misc/prio3_multiproof_robustness.py

```diff
@@ -0,0 +1,100 @@
+# prio3_multiproof_robustness.py - Plot robustness bounds for various parameters.
+# Use `sage -python prio3_multiproof_robustness.py`
+import matplotlib.pyplot as plt
+import math
+
+FIELD128_MODULUS = 2**66 * 4611686018427387897 + 1  # Field128.MODULUS
+FIELD64_MODULUS = 2**32 * 4294967295 + 1            # Field64.MODULUS
+
+NUM_REPORTS = 1000000000
+
+
+def soundness(gadget_calls, gadget_degree, field_size):
+    '''
+    ia.cr/2019/188, Theorem 4.3
+
+    gadget_calls - number of times the gadget is called
+
+    gadget_degree - arithmetic degree of the gadget
+
+    field_size - size of the field
+    '''
+    return gadget_calls * gadget_degree / (field_size - gadget_calls)
+
+
+def robustness(epsilon, ro_queries, prep_queries, num_proofs):
+    '''
+    ia.cr/2023/130, Theorem 1, ignoring negligible terms and assuming the bound
+    can be modified by raising `epsilon` to the power of the number of FLPs.
+
+    epsilon - soundness of the base FLP
+
+    ro_queries - random oracle queries, a proxy for the amount of precomputation
+                 done by the adversary
+
+    prep_queries - number of online attempts, a proxy for the batch size
+
+    num_proofs - number of FLPs
+    '''
+    return (ro_queries + prep_queries) * epsilon**num_proofs
+
+
+def sum_vec(field_size, num_proofs, length):
+    '''
+    Prio3SumVec (draft-irtf-cfrg-vdaf-08, Section 7.4.3): Probability of
+    accepting one report in a batch of NUM_REPORTS. Assuming the asymptotically
+    optimal chunk length.
+    '''
+
+    # Table 11
+    def gadget_calls(length, bits, chunk_length):
+        return (length * bits + chunk_length - 1) // chunk_length
+
+    return robustness(
+        soundness(
+            gadget_calls(length, 1, max(1, math.sqrt(length))),
```

Comment:
nit: `math.sqrt` will produce a float, better to use round down or up? since `chunk_length` should be an integer.